src/crimson/osd/osd_operations/background_recovery.h

```diff
@@ -0,0 +1,46 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include "crimson/net/Connection.h"
+#include "crimson/osd/osd_operation.h"
+#include "crimson/common/type_helpers.h"
+
+class MOSDOp;
+
+namespace crimson::osd {
+class PG;
+class ShardServices;
+
+class BackgroundRecovery final : public OperationT<BackgroundRecovery> {
+  Ref<PG> pg;
+  ShardServices &ss;
+  epoch_t epoch_started;
+  crimson::osd::scheduler::scheduler_class_t scheduler_class;
+
+  auto get_scheduler_params() const {
+    return crimson::osd::scheduler::params_t{
+      1, // cost
+      0, // owner
+      scheduler_class
+    };
+  }
+
+  seastar::future<bool> do_recovery();
```

Comment:
could you move the private members to the end of class?

Reply:
I really don't remember this being part of our style guidelines.  Many many bits portions of the ceph code base put private elements first.

Reply:
see https://google.github.io/styleguide/cppguide.html#Declaration_Order , personally, i feel that it'd be easier to read a class if its interface comes first in the header.

> Many many bits portions of the ceph code base put private elements first.

yeah, my guess is that many of us had with a strong history of C where a function/variable should be defined literary *before* where it's used.